ui/src/contexts/Identity.tsx

```diff
@@ -0,0 +1,175 @@
+import { message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import { getIdentities, identifierParser } from "src/adapters/api/identities";
+import { useEnvContext } from "src/contexts/Env";
+import { AppError, Identity } from "src/domain";
+import { ROUTES } from "src/routes";
+import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import {
+  getStorageByKey,
+  isAbortedError,
+  makeRequestAbortable,
+  setStorageByKey,
+} from "src/utils/browser";
+import {
+  IDENTIFIER_LOCAL_STORAGE_KEY,
+  IDENTIFIER_SEARCH_PARAM,
+  ROOT_PATH,
+} from "src/utils/constants";
+import { buildAppError } from "src/utils/error";
+
+type IdentityState = {
+  fetchIdentities: (signal: AbortSignal) => void;
+  handleChange: (identifier: string) => void;
+  identifier: string;
+  identityDisplayName: string;
+  identityList: AsyncTask<Identity[], AppError>;
+};
+
+const defaultIdentityState: IdentityState = {
+  fetchIdentities: () => void {},
+  handleChange: () => void {},
+  identifier: "",
+  identityDisplayName: "",
+  identityList: { status: "pending" },
+};
+
+const IdentityContext = createContext(defaultIdentityState);
+
+export function IdentityProvider(props: PropsWithChildren) {
+  const env = useEnvContext();
+  const [messageAPI, messageContext] = message.useMessage();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [identityList, setIdentityList] = useState<AsyncTask<Identity[], AppError>>({
+    status: "pending",
+  });
+  const [identifier, setIdentifier] = useState("");
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const identity =
+    (identityList.status === "successful" || identityList.status === "reloading") &&
+    identityList.data.find((identity) => identity.identifier === identifier);
+  const identityDisplayName = identity && identity.displayName ? identity.displayName : "";
```

Comment:
Again, we're failing to properly fallback the displayName. Let's change this because it's obviously problematic.

Instead of falling back to the short did everywhere, let's do it once, in the adapter, so the view does not need to worry about this.

This means the displayName is required in our domain entity, and we do the fallback only once and forget about it everywhere else.

Reply:
we don't need to fall back to short did here, because it does not make sense to show a short did twice